code/game/objects/structures/crates_lockers/crates.dm

```diff
@@ -23,9 +23,11 @@
 	drag_slowdown = 0
 	var/obj/item/paper/fluff/jobs/cargo/manifest/manifest
 	var/radius_2 = 1.35
+	var/animation_math
```

Comment:
```suggestion
	var/static/list/animation_math
```

Reply:
I understand what you mean only problem is this would bring some problems for any crate that you want to have other values for things like the animation angle or azimuth angle or another animation time as those would share the list for items with these specific values now this is no problem at the moment but could introduce some headache later honestly i do not like the use of lists inside lists anyways i would more like to use a 2 dimensional matrix instead of a list

Reply:
but i think i can reduced the total ammount of lists generated to a single one for every obj without reducing the ammount of actual values

Reply:
Wait i have an idea that might reduce the amount of lists more if i can create a static list for every child obj of crates so that all child objects of the same type share that list but i am not sure how to best go about that

Reply:
i think i can pull this off i will create a static list and for every child object a list inside with the first index beeing used to identify its obj path thanks for that idea its still a few more lists but it will greatly reduce the total amount of actual lists

Reply:
If you feel *really* brave, make it a static assoc list of lists with animation-determining parameters as a key so every crate with the same animation parameters can share a single pre-calculated animation list, e.g.:
`animation_list["[door_anim_time]-[door_anim_angle]-[door_hinge]-[radius_2]-[open_angle]"][idx]`

Reply:
Yes did it thanks for that idea again got a assoziative list that gives back the needed index for the animation_math list and thus only creates a new list if there is actually a crate that gets initialized with a new set of values(or if an admin vv's a crate and calls the proc)